src/plots/gl2d/camera.js

```diff
@@ -76,32 +99,114 @@ function createCamera(scene) {
+                        result.boxStart[1] !== result.boxEnd[1])
+                    ) {
+                        result.boxEnabled = true;
+                    }
+
+                    // constrain aspect ratio if the axes require it
+                    var smallDx = Math.abs(result.dragStart[0] - x) < MINZOOM;
+                    var smallDy = Math.abs(result.dragStart[1] - y) < MINZOOM;
+                    if(getSubplotConstraint() && !(smallDx && smallDy)) {
+                        dx = result.boxEnd[0] - result.boxStart[0];
+                        dy = result.boxEnd[1] - result.boxStart[1];
+                        var dydx = (dataBox[3] - dataBox[1]) / (dataBox[2] - dataBox[0]);
+
+                        if(Math.abs(dx * dydx) > Math.abs(dy)) {
+                            result.boxEnd[1] = result.boxStart[1] +
+                                Math.abs(dx) * dydx * (Math.sign(dy) || 1);
+
+                            // gl-select-box clips to the plot area bounds,
+                            // which breaks the axis constraint, so don't allow
+                            // this box to go out of bounds
+                            if(result.boxEnd[1] < dataBox[1]) {
+                                result.boxEnd[1] = dataBox[1];
+                                result.boxEnd[0] = result.boxStart[0] +
+                                    (dataBox[1] - result.boxStart[1]) / Math.abs(dydx);
+                            }
+                            else if(result.boxEnd[1] > dataBox[3]) {
+                                result.boxEnd[1] = dataBox[3];
+                                result.boxEnd[0] = result.boxStart[0] +
+                                    (dataBox[3] - result.boxStart[1]) / Math.abs(dydx);
+                            }
+                        }
+                        else {
+                            result.boxEnd[0] = result.boxStart[0] +
+                                Math.abs(dy) / dydx * (Math.sign(dx) || 1);
+
+                            if(result.boxEnd[0] < dataBox[0]) {
+                                result.boxEnd[0] = dataBox[0];
+                                result.boxEnd[1] = result.boxStart[1] +
+                                    (dataBox[0] - result.boxStart[0]) * Math.abs(dydx);
+                            }
+                            else if(result.boxEnd[0] > dataBox[2]) {
+                                result.boxEnd[0] = dataBox[2];
+                                result.boxEnd[1] = result.boxStart[1] +
+                                    (dataBox[2] - result.boxStart[0]) * Math.abs(dydx);
+                            }
+                        }
+                    }
+                    // otherwise clamp small changes to the origin so we get 1D zoom
+                    else {
+                        if(smallDx) result.boxEnd[0] = result.boxStart[0];
+                        if(smallDy) result.boxEnd[1] = result.boxStart[1];
```

Comment:
@etpinard while I was at it I added in a few of the bits of behavior used in svg, such as 1D zoom, minimum zoom size, and clamping panning to pure x (y) if dy (dx) is small enough. Do you want to take it for a spin and see what you think?